components/nimbus/src/evaluator.rs

```diff
@@ -382,7 +387,7 @@ fn query_event_store(
         }
     } as usize;
 
-    Ok(json!(event_store.query(
+    Ok(json!(event_store.lock().unwrap().query(
```

Comment:
This is the only time that the EventStore is actually pulled out of the Arc<Mutex<>> and used, can we can pass a reference to the Arc<Mutex<>> all the way to this point to reduce the number of times that it is cloned?